rfqmsg/accept.go

```diff
@@ -0,0 +1,225 @@
+package rfqmsg
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// Accept message type field TLV types.
+
+	TypeAcceptID          tlv.Type = 0
+	TypeAcceptAskingPrice tlv.Type = 2
+	TypeAcceptExpiry      tlv.Type = 4
+	TypeAcceptSignature   tlv.Type = 6
+)
+
+func TypeRecordAcceptID(id *ID) tlv.Record {
+	return tlv.MakePrimitiveRecord(TypeAcceptID, id)
+}
+
+func TypeRecordAcceptAskingPrice(askingPrice *lnwire.MilliSatoshi) tlv.Record {
+	return tlv.MakeStaticRecord(
+		TypeAcceptAskingPrice, askingPrice, 8, milliSatoshiEncoder,
+		milliSatoshiDecoder,
+	)
+}
+
+func milliSatoshiEncoder(w io.Writer, val interface{}, buf *[8]byte) error {
+	if ms, ok := val.(*lnwire.MilliSatoshi); ok {
+		return tlv.EUint64(w, uint64(*ms), buf)
+	}
+
+	return tlv.NewTypeForEncodingErr(val, "MilliSatoshi")
+}
+
+func milliSatoshiDecoder(r io.Reader, val interface{}, buf *[8]byte,
+	l uint64) error {
+
+	if ms, ok := val.(*lnwire.MilliSatoshi); ok {
+		var msInt uint64
+		err := tlv.DUint64(r, &msInt, buf, l)
+		if err != nil {
+			return err
+		}
+
+		*ms = lnwire.MilliSatoshi(msInt)
+		return nil
+	}
+
+	return tlv.NewTypeForDecodingErr(val, "MilliSatoshi", l, 8)
+}
+
+func TypeRecordAcceptExpiry(expirySeconds *uint64) tlv.Record {
+	return tlv.MakePrimitiveRecord(
+		TypeAcceptExpiry, expirySeconds,
+	)
+}
+
+func TypeRecordAcceptSig(sig *[64]byte) tlv.Record {
+	return tlv.MakePrimitiveRecord(
+		TypeAcceptSignature, sig,
+	)
+}
+
+// acceptMsgData is a struct that represents the data field of a quote
+// accept message.
+type acceptMsgData struct {
+	// ID is the unique identifier of the request for quote (RFQ).
+	ID ID
+
+	// AskingPrice is the asking price of the quote.
+	AskingPrice lnwire.MilliSatoshi
+
+	// Expiry is the asking price expiry lifetime unix timestamp.
+	Expiry uint64
```

Comment:
If you use the new `tlv` stuff, then this can just be: 
```go
Expiry tlv.RecordT[tlv.Type4, uint64]
```

and so forth

Then you don't need to make thing like `TypeRecordAcceptID`, as those types _are_ records themselves (or rather can become one).

Reply:
Goes for all the new TLV related structs here. 

I think this also needs an LN message framing layer as well. So the wire messages that can be decoded by something using the BOLT protocol message headers.